Fix bug 14634: symbolic permissions causes error

tonk opened the pull request:
##### Issue Type:
- Bugfix Pull Request
##### Ansible Version:

```
1.9.4 and 2.0.1.0 and 2.1.0
```
##### Summary:

Fix bug 14634: symbolic permissions causes error

Commit:
cloudstack: add tests for cs_volume

bcoca committed:
common retry and rate limiting decorators for apis

allows modules to wrap their request functions in common retry and rate limiting
decorators or functions

resmo committed:
cloudstack: cs_volume: add tests for volume resize

See https://github.com/ansible/ansible-modules-extras/pull/1333

resmo committed:
cloudstack: new integration tests test_cs_configuration

resmo committed:
cloudstack: new integration tests test_cs_instance_facts

resmo committed:
cloudstack: new integration tests test_cs_pod

resmo committed:
cloudstack: new integration tests test_cluster

resmo committed:
cloudstack: new integration tests test_cs_resourcelimit

Commit:
Issue #14634: symbolic permissions error

chmod with symbolic permissions string causes error

This fixes the symbolic notation of the chmod modes,
as stated in the man page of chmod (in Linux).
This also takes into account that ` chmod a+x` is
different from `chmod +x`. As the second one
should thake the current umask into account.

Commit:
with self

Commit:
raise if invalid.

Commit:
Ternary statement removed. Doesn't work in 2.4.

Commit:
I better quit. It's not my day :-)

Commit:
initial checkin of the new Community Committer Guidelines

Commit:
fix formatting for people section

bcoca committed:
paging for amazon modules

this adds a decorator for use in amazon modules so retrieval functions
can page using the 'marker' that most boto functions return

mattclay committed:
Fix unicode handling in connection plugins.

mattclay committed:
Add additional plugins to connection tests.

- docker
- libvirt_lxc
- jail

These tests will not run as part of the build, but can be run manually.

AevaOnline committed:
Send empty command before recv

On some switches, starting an SSH connection and immediately calling
recv() will result in a connection timeout. The switch requires some
input on the channel before it provides any prompt.

As such, this patch sends an empty command immediately upon connection,
triggering the switch to send a prompt which the shell can then
interpret.

Signed-off-by: Devananda van der Veen <devananda.vdv@gmail.com>

Commit:
Merge branch 'Fix_Issue_14634' into devel

* Fix_Issue_14634:
  Issue #14634: symbolic permissions error

amenonsen commented:
How about squashing to get rid of the four unnecessary commits and rebasing to get rid of the merge?

resmo committed:
docsite, cloudstack: add limited VPC support

Also see https://github.com/ansible/ansible-modules-extras/issues/1811

jimi-c committed:
Removing explicit setting of failed/failed_when

dagwieers committed:
Move the Github issue and pull-request templates to .github

In essence, most people do not need to use the templates directly,
getting them out of the way increases the signal-to-noise ratio of
the root directory. Direct people to what they are looking for.

jimi-c committed:
Revert "Removing explicit setting of failed/failed_when"

This reverts commit 4e528e9535732b7541155b323cdab2377d74cc3a.

jimi-c committed:
Revert "complete correction of test, key exists but is false"

This reverts commit 7e6343213b9ba78db828733faf9d9c0cb8b96ece.

jimi-c committed:
Revert "corrected failed_when test"

This reverts commit 140df20504617dcf5f7e4951443b9110420d4792.

bcoca committed:
avoid creating extra tmp dirs we don't delete

fixes #14867

bcoca committed:
allow tests to run with parameters

also fixed test_test_infra to allow tags

jimi-c committed:
Don't always insert failed in the results (again)

jimi-c committed:
Adding some packages to the unit test portions of travis testing

Missing python-memcached and python-redis result in some tests being
skipped and lowering the overall coverage %

jimi-c committed:
Fixing memcached->python-memcached for pip install of travis setup

jimi-c committed:
Fixing typo from python-memcached change in travis yml

jimi-c committed:
Moving pip test requirements from travis.yml to tox test-requirements.txt

jimi-c committed:
Also adding python-systemd to tox test-requirements.txt

jimi-c committed:
Fixing template/assemble action plugins related to tmp dir use/cleanup

jimi-c committed:
Removing forced pull from build step in run_tests.sh

jimi-c committed:
Updating RELEASES/etc. for devel

[ci skip]

jimi-c committed:
Add mysql connection_timeout param to module_utils/mysql.py

bcoca committed:
predictable docker names and autoremove if no fail

mattclay committed:
Lower connect timeout for test_mysql_variables.

jimi-c committed:
Fixing equality check in run_tests.sh for sanity

amesbury committed:
Fixed typo

mattclay committed:
Add noseclabel support to libvirt_lxc plugin.

Commit:
bugfix for shared module shell.py

removes get_cli_connection function which was left over from a refactor
but was no longer in use

jimi-c committed:
Moving docker tests to pre-compiled images

Commit:
Set type for OpenStack 'verify' param

The 'verify' param is a bool, so we don't want it to be an assumed
str.

jimi-c committed:
Reorganizing tox stuff and making py3-specific requirements

Commit:
updates based on feedback from docschick and bcoca.

dagwieers committed:
Reformat new module lists

Make the Changelog easier to parse, especially the list of new modules

mattclay committed:
Add unicode dir to connection tests.

mattclay committed:
Add missing to_bytes on directory path.

Commit:
bugfix for eos shared module and new optimization

This commit address to issues in the eos shard module.  The first one
is a bug fix for returning the running config when the transport is eapi.
The shared module will now return config text instead of an object.  The
second is a optimization that delays when the eos module connects to the
remote devices.  This provies a performance enhancement when using
ssh since the module doesn't default to connecting immediately

Yannig committed:
Add a way to restrict gathered facts in Ansible:
- Using gather_subset options
- By ignoring ohai/chef or facter/puppet facts

jimi-c committed:
Initial commit for Ubuntu 12.04 docker config for testing

jimi-c committed:
Adding ubuntu1204 to the travis config for testing

mscherer committed:
Fix the lexer used, jinja2 is not valid, jinja is

jimi-c committed:
Updating jinja2 in ubuntu1204 docker image

jimi-c committed:
Fixing typo in ubuntu1204 docker config

jimi-c committed:
Adding rubygems to the ubuntu1204 docker config

[no ci]

jimi-c committed:
Also adding an upgrade of pycrypto to ubuntu1204 config

jimi-c committed:
Adding python-dev to the list of packages for ubuntu1204 docker

[ci skip]

Commit:
feature to allow prompts to be configured at instatiation

This commit adds a new feature to allow implementations of shell to
specify the command prompt regexp to be used.  It allows adds a new
kwarg at instantiation to kick the remote device with a carriage return.
By default the kickstart flag is true but can be disabled by passing
kickstart=False.

Commit:
feature in ios to tell shell not to kickstart

This commit is necessary to tell shell not to kickstart the cli session as
it causes problems in IOS to recognize the prompt.

Commit:
feature to implement localized cli prompts to eos

This change localizes the cli prompt regexps to the eos shared module
instead of the common prompts implemented in shell

jimi-c committed:
Adding unit tests for TaskResult

abadger committed:
Make changes proposed during review of restrict fact gathering feature:

* Make documentation examples into code blocks
* Make code to call the subsets more general.
* Made min subset always execute (cannot disable it).
* Use a passed in modules parameter rather than global modules.  This is needed for ziploader
* Remove unneeded __init__()
* Remove uneeded multiple inheritance from a base class
* gather_facts is now a list type

bcoca committed:
now item callback honors display_skipped_hosts

fixes #14956

abadger committed:
Allow FieldAttribute lists and sets to be set from a comma separated string

abadger committed:
Make ohai and facter work via module_utils Fact classes rather than in the setup module

Commit:
updates to nxos shared module

This commit address a number of minor updates the nxos shared module

* connect() is now lazy loaded
* parse inner output messages when errored
* code syntax cleanup

Commit:
fixes ansible-modules-code#3250

Sets the default value for transport to 'cli'

abadger committed:
Make integration tests for fact gathering assert on failure

bcoca committed:
added warning about additional requirements

bcoca committed:
don't raise exceptoins on bad hosts files

fixes #14969

holmboe committed:
2016 is the year

abadger committed:
Update submodule refs

abadger committed:
add a fact gathering check for the default of all

jimi-c committed:
Adding iproute to certain RH-based docker images

abadger committed:
Make the fact gathering give a little more information on failure

resmo committed:
changelog: append cs_zone_facts

abadger committed:
Fix saving of tracebacks on module failure

tonk commented:
What a mess. I'm really starting to hate Git!

tonk commented:
I'll resubmit
